src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/HealthCheckFunctions.cs

```diff
@@ -1,63 +1,82 @@
 using GovUk.Education.ExploreEducationStatistics.Public.Data.Model.Database;
-using GovUk.Education.ExploreEducationStatistics.Public.Data.Services.Options;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Services.Interfaces;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
 using Microsoft.Azure.Functions.Worker;
-using Microsoft.Azure.Functions.Worker.Http;
 using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.Logging;
-using Microsoft.Extensions.Options;
 
 namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Functions;
 
 // ReSharper disable once ClassNeverInstantiated.Global
 public class HealthCheckFunctions(
     ILogger<HealthCheckFunctions> logger,
     PublicDataDbContext publicDataDbContext,
-    IOptions<ParquetFilesOptions> parquetFileOptions)
+    IDataSetVersionPathResolver dataSetVersionPathResolver)
 {
-    [Function(nameof(CountDataSets))]
-    public async Task<string> CountDataSets(
+    [Function(nameof(HealthCheck))]
+    [Produces("application/json")]
+    public async Task<IActionResult> HealthCheck(
 #pragma warning disable IDE0060
-        [HttpTrigger(AuthorizationLevel.Function, "get")] HttpRequestData request)
+        [HttpTrigger(AuthorizationLevel.Anonymous, "get")] HttpRequest request)
 #pragma warning restore IDE0060
     {
+        var psqlConnectionHealthCheck = await CheckPsqlConnectionHealthy();
+        var fileShareMountHealthCheck = CheckFileShareMountHealth();
+        var healthCheckResponse = new HealthCheckResponse(
+            PsqlConnection: psqlConnectionHealthCheck,
+            FileShareMount: fileShareMountHealthCheck);
```

Comment:
We've also got connections on the content database and on blob storage, so I'm wondering if you considered also making those part of the health check?

Reply:
Added in thanks!

Reply:
👍 Just wondering if you are planning to add more tests to `HealthCheckFunctionTests` as currently the only failure variant is for the missing file share? No worries if not though.

Reply:
I couldn't think how to provide failures in the other cases without a decent element of overengineering to make a DbContext fail or shut down storage prior to calling it :)  I guess I *could* just shut down storage, but the DbContexts I reckon would be fiddly!